### PR TITLE
Client.AddSequencedLeafAndWait

### PR DESCRIPTION
**Pull request by gdbelvin**
Adds `AddSequencedLeafAndWait` to the client and uses it in some tests.

### CODE REVIEW

File: client/log_client_test.go [hunk 1]
```diff
@@ -36,13 +37,15 @@ func TestAddGetLeaf(t *testing.T) {
 // addSequencedLeaves is a temporary stand-in function for tests until the real API gets built.
 func addSequencedLeaves(ctx context.Context, env *integration.LogEnv, client *LogClient, leaves [][]byte) error {
 	// TODO(pavelkalinnikov): Replace with AddSequencedLeaves batch API.
```
Comment (pavelkalinnikov) [inline client/log_client_test.go]: You can remove this TODO, I think.

File: client/log_client_test.go [hunk 2]
```diff
@@ -36,13 +37,15 @@ func TestAddGetLeaf(t *testing.T) {
...
-		if err := client.QueueLeaf(ctx, l); err != nil {
-			return err
+	for i, l := range leaves {
+		if err := client.AddSequencedLeaf(ctx, l, int64(i)); err != nil {
+			return fmt.Errorf("AddSequencedLeaf(): %v", err)
 		}
-		env.Sequencer.OperationSingle(ctx)
+	}
+	env.Sequencer.OperationSingle(ctx)
+	for _, l := range leaves {
```
Comment (pavelkalinnikov) [inline client/log_client_test.go]: Would just waiting for N-th entry inclusion work? Looks like it implies other entries are there as well.

File: client/log_client_test.go (hunk 1 above)
Comment (gdbelvin) [reply client/log_client_test.go]: Done

File: client/log_client_test.go (hunk 2 above)
Comment (gdbelvin) [reply client/log_client_test.go]: Done

**Pull request merged**